### ISSUE TITLE
[1,5,0,1,0,1] is frettable

### ISSUE THREAD

**Issue by hyvyys** [yomed/frettable#1]
I tested the package with just this one combination, and it returned true. Those are some mad skills!

**Comment by yomed** [comment]
Hi @hyvyys, thanks for checking out the project!

Right now, the heuristics are around whether there are enough fingers and the notes are in the right spots such that it can be fretted/barred (given fingers of any length...). It could be a nice improvement to limit how "wide" the chord can be, maybe based on the part of the neck too. E.g. the chords near the lower frets could only span ~4 frets, while the higher fretted chords could span more than that.

**Comment by hyvyys** [comment]
Is a really interesting problem. I tried do code this for my chord
fingering generator and I noticed that four fret range is playable
(supposing those Voldemort fingers) but only when the distribution of
fingers is, well, anatomical. They need to go in equal distances and not
like 104101 or 404104.

Do you consider any single frets pressed *below* a partial barre playable?
I.e. is it humane to press a short barre with fingers other than index or
index and middle, idk, pinky?

śr., 9 paź 2019, 05:29 użytkownik Yoni Medoff <notifications@github.com>
napisał:

**Comment by yomed** [comment]
Right. I think one way to get around the Voldemort finger problem would be to start off categorizing chords into physically possible and physically impossible, and then create a difficulty score associated for those chords that are possible. I think the examples you pointed out are good options for some heuristics. I think that depending on the use case, it may or may not be appropriate to generate chords based on some difficulty threshold. For use cases leaning toward advanced usage, I would consider the pinky finger to be able to barre pretty well.

I had attempted to make some rules around barring, though it's been a while since I've reviewed the logic:
https://github.com/yomed/frettable/blob/64e9018eb5c5bdc7cca941073d7d9af45992b3b4/index.js#L19-L22
